### PR TITLE
[Merged by Bors] - Make it explicit that we expect address as an input for genesis accounts

### PR DESCRIPTION
**Pull request by dshulyak**
## Motivation

types.BytesToAddress will return only the rightmost 20 bytes if value is larger than 20 bytes, but to be explicit we will return an error if the provided value is not address but a public key.

related: https://github.com/spacemeshos/go-spacemesh/issues/2493

## Changes
- return an error if provided address is less than expected size

## Test Plan
unit tests, system tests

## TODO

- [ ] Explain motivation or link existing issue(s)
- [ ] Test changes and document test plan
- [ ] Update documentation as needed

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Review by nkryuchkov — APPROVED** [review]

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

File: cmd/node/node.go [hunk 1]
```diff
@@ -399,14 +399,17 @@ func (app *App) setupGenesis(state *state.TransactionProcessor, msh *mesh.Mesh)
 	for id, balance := range app.Config.Genesis.Accounts {
 		bytes := util.FromHex(id)
 		if len(bytes) == 0 {
-			return fmt.Errorf("cannot read config entry for genesis account %s", id)
+			return fmt.Errorf("cannot decode entry %s for genesis account", id)
+		}
+		// just make it explicit that we want address and not a public key
+		if len(bytes) > types.AddressLength {
```
Comment (countvonzero) [inline cmd/node/node.go]: `!=` to be precise?

**Review by countvonzero — APPROVED** [review]

File: cmd/node/node.go (hunk 1 above)
Comment (dshulyak) [reply cmd/node/node.go]: there was 0x1 in default config file, and i didn't want to add 19 zeroes but i think i can remove it completely

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**